### PR TITLE
OPENEUROPA-1907: Add oembed routing.

### PR DESCRIPTION
**Pull request by imanoleguskiza**

### CODE REVIEW

File: src/Controller/OembedRouteController.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+declare(strict_types = 1);
+
+namespace Drupal\oe_oembed\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * Oembed route controlled.
```
Comment (upchuk) [inline src/Controller/OembedRouteController.php]: controller.

File: tests/Browser/OembedRouteTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\Tests\oe_oembed\Browser;
+
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests regarding the responses gotten from the oembed route.
```
Comment (upchuk) [inline tests/Browser/OembedRouteTest.php]: Tests the oEmbed provider responses.

File: tests/Browser/OembedRouteTest.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class OembedRouteTest extends BrowserTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
```
Comment (upchuk) [inline tests/Browser/OembedRouteTest.php]: The parent is "protected" so it needs to be protected.

File: tests/Browser/OembedRouteTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    'oe_oembed',
+  ];
+
+  /**
+   * Tests oembed route returns a correct json.
```
Comment (upchuk) [inline tests/Browser/OembedRouteTest.php]: Tests that the oEmbed route returns the correct JSON.

File: tests/Browser/OembedRouteTest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+   */
+  public function testOembedRoute() {
+
+    $this->drupalGet('oembed');
+    $this->assertSession()->statusCodeEquals(200);
+
+    // Test that there is an empty reaction rule listing.
```
Comment (upchuk) [inline tests/Browser/OembedRouteTest.php]: No it doesn't test that.

File: src/Controller/OembedRouteController.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Oembed route controller.
+ */
+class OembedRouteController {
+
+  /**
+   * Returns the oEmbed json object related to the request.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   The access result.
```
Comment (brummbar) [inline src/Controller/OembedRouteController.php]: It's not the access result being returned.

File: src/Controller/OembedRouteController.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+   */
+  public function getOembedJson() {
```
Comment (brummbar) [inline src/Controller/OembedRouteController.php]: Add the return type hint.

File: src/Controller/OembedRouteController.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  /**
+   * Checks access for a specific request.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Run access checks for this account.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function checkAccess(AccountInterface $account) {
```
Comment (brummbar) [inline src/Controller/OembedRouteController.php]: Return type hint missing.

File: tests/Browser/OembedRouteTest.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the oEmbed provider responses.
+ */
+class OembedRouteTest extends BrowserTestBase {
+
+  /**
+   * Modules to enable.
```
Comment (brummbar) [inline tests/Browser/OembedRouteTest.php]: Use `{@inheritdoc}` here.

File: tests/Browser/OembedRouteTest.php [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+   */
+  protected static $modules = [
+    'oe_oembed',
+  ];
+
+  /**
+   * Tests that the oEmbed route returns the correct json.
+   */
+  public function testOembedRoute() {
+
```
Comment (brummbar) [inline tests/Browser/OembedRouteTest.php]: No empty line under function declaration.

**Pull request merged**